mlrun/datastore/sources.py

```diff
@@ -73,32 +73,46 @@ def get_table_object(self):
         """get storey Table object"""
         return None
 
-    def to_dataframe(self):
-        return mlrun.store_manager.object(url=self.path).as_df()
-
-    def filter_df_start_end_time(self, df, time_field):
-        # give priority to source time_field over the feature set's timestamp_key
-        if self.time_field:
-            time_field = self.time_field
-
-        if self.start_time or self.end_time:
-            self.start_time = (
-                datetime.min if self.start_time is None else self.start_time
-            )
-            self.end_time = datetime.max if self.end_time is None else self.end_time
-            df = df.filter(
-                (df[time_field] > self.start_time) & (df[time_field] <= self.end_time)
-            )
-        return df
+    def to_dataframe(
+        self,
+        columns=None,
+        df_module=None,
+        entities=None,
+        start_time=None,
+        end_time=None,
+        time_field=None,
+    ):
+        """return the source data as dataframe"""
+        return mlrun.store_manager.object(url=self.path).as_df(
+            columns=columns,
+            df_module=df_module,
+            start_time=start_time or self.start_time,
+            end_time=end_time or self.end_time,
+            time_column=time_field or self.time_field,
+        )
 
-    def to_spark_df(self, session, named_view=False, time_field=None):
+    def to_spark_df(self, session, named_view=False, time_field=None, columns=None):
         if self.support_spark:
             df = session.read.load(**self.get_spark_options())
             if named_view:
                 df.createOrReplaceTempView(self.name)
-            return df
+            return self._filter_spark_df(df, time_field, columns)
         raise NotImplementedError()
 
+    def _filter_spark_df(self, df, time_field=None, columns=None):
+
+        from pyspark.sql.functions import col
```

Comment:
I think it might be better to start by checking and if both `columns` and `time_field` are not defined simply return `df` as is. This will save importing for nothing.